packages/edit-post/src/components/options-modal/options/enable-custom-fields.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { withSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import BaseOption from './base';
+
+class EnableCustomFieldsOption extends Component {
+	constructor( { settings: { isEnabled } } ) {
+		super( ...arguments );
+
+		this.toggleCustomFields = this.toggleCustomFields.bind( this );
+
+		this.state = { isEnabled };
+	}
+
+	toggleCustomFields() {
+		const { toggleURL, isEnabled } = this.props.settings;
+
+		// Redirect to an admin action that toggles the setting and reloads the editor
+		// with the meta box assets included on the page.
+		window.location.href = toggleURL;
```

Comment:
Such a hack!

Reply:
Yeah, not proud of this! 🙃

Reply:
Thinking it might be nicer to run the action by calling `.submit()` on a hidden `<form>` that's on the page. That way, we'll trigger the action using a `POST` request instead of a `GET` request. (Not that we seem to mind in WP Admin, as e.g. the _Trash_ link deletes posts using `GET`.)

Reply:
I've done this in 6a18bcdb9f8e16135e5c15f595f18ab254f4d06b. I hate it a little less, now.